cmd/client/main.go

```diff
@@ -30,14 +31,30 @@ func main() {
 		os.Exit(0)
 	}
 
-	scn := scanner.FromStr(flg.scanner)
+	notools := scanner.NoTools()
+
+	if notools {
+		log.Println("[Core] No scanner tools available")
+		os.Exit(0)
+	}
+
+	scn, found := scanner.FromStr(flg.scanner)
+
+	if flg.scanner == "" {
+		defaultTool := scanner.Default
+		log.Println("[Core] No scanner tool specified, default scanner tool " + defaultTool.Name + " is selected")
+		scn = defaultTool
+	} else if !found {
+		log.Println("[Core] Requested scanner tool not found. Use option -l to view available scanner tools")
+		os.Exit(0)
+	}
+
```

Comment:
The `if-else` block is not really clean. 

You should change it to 

```go
if notTools {
...
}

scn, found := scanner.FromStr(flg.scanner)
if !found {
   scn = scanner.Default
   log.Printf("[Core] Unable to find scanner with name '%s', instead using default scanner", scn.Name)
}

log.Printf("[Core] Use scanner '%s'", scn.Name)

```

Reply:
What if user is not interested in the default scanner or mistyped the name of the desired tool?

Reply:
But I agree the code is not very clean

Reply:
The code got better, but the last `if` statement is useless. 
I see two options.
(1) remove the `if` statement in the middle and cancel the app if the scanner could not be found.
(2) remove the last `if` statement and proceed with the default scanner.

Reply:
Instead of 
`log.Println("somthing" + value + "something" )`

use 
```
log.Println("Value is  :", myValue)
// or
log.Printf("Value is  : %g", myValue)
```

Reply:
If I remove the middle if statement and no tool is specified (which implies that user is interested in default scanner), the program shuts down. If I remove the last statement and a tool name is mistyped, the user may get confused
These were my concerns when changing the code, I go with the second option in my next commit